pkg/controller/account/byoc.go

```diff
@@ -362,3 +337,47 @@ func validateBYOCClaim(accountClaim *awsv1alpha1.AccountClaim) error {
 
 	return nil
 }
+
+func deleteOtherAccessKeys(reqLogger logr.Logger, client awsclient.Client, accessKeyID string) error {
+
+	accessKeyList, err := client.ListAccessKeys(&iam.ListAccessKeysInput{})
+	for _, accessKey := range accessKeyList.AccessKeyMetadata {
+		if *accessKey.AccessKeyId != accessKeyID {
+			_, err = client.DeleteAccessKey(&iam.DeleteAccessKeyInput{AccessKeyId: accessKey.AccessKeyId})
+			if err != nil {
+				reqLogger.Error(err, "Failed to delete BYOC access keys")
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func replaceSecretAccessKey(reqLogger logr.Logger, client awsclient.Client, secret corev1.Secret, userName string, accessKeyID *string) (corev1.Secret, error) {
```

Comment:
Please leave a comment about what this function does. The name suggest to me it just deals with the secret, but the function creates and deletes IAM resources as well.

Reply:
@fahlmant I might be confused but added a comment with what I believe this method does

Reply:
Maybe a better name for this function would be `RotateIAMUserKeys`?

Reply:
I had a similar thought, but this function rotates the keys AND places it in a secret. Ideally this would be two functions.